joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/RubyTypeRecoveryTests.scala

```diff
@@ -182,20 +183,35 @@ class RubyExternalTypeRecoveryTests
         |
         |def func
         |   sg = SendGrid::API.new(api_key: ENV['SENDGRID_API_KEY'])
-        |   sg2 = SendGrid::MailSettingsDto.fetch
         |   response = sg.client.mail._('send').post(request_body: data)
         |end
         |""".stripMargin,
       "main.rb"
     )
       .moreCode(RubyExternalTypeRecoveryTests.SENDGRID_GEMFILE, "Gemfile")
 
-    // TODO: Should be fixed when external dependency solver is done
-    "be present in (Case 1)" ignore {
-      cpg.identifier("sg").lineNumber(5).typeFullName.l shouldBe List(
-        "sendgrid/sendgrid.rb:<global>::program.SendGrid.API"
-      )
-      cpg.call("client").methodFullName.l shouldBe List("sendgrid/sendgrid.rb:<global>::program.SendGrid.API:client")
+    "be present in (Case 1)" in {
+      cpg.identifier("sg").lineNumber(5).typeFullName.l shouldBe List("sendgrid-ruby.SendGrid.API")
+      cpg.call("client").methodFullName.l shouldBe List("sendgrid-ruby.SendGrid.API:client")
+    }
+
+    "resolve correct imports via tag nodes" in {
+      inside(cpg.call.where(_.referencedImports).l) {
+        case sendgridImport :: Nil =>
+          inside(
+            sendgridImport.tag._toEvaluatedImport
+              .filter(_.label == EvaluatedImport.RESOLVED_METHOD)
+              .map(_.asInstanceOf[ResolvedMethod])
+              .filter(_.fullName.startsWith("sendgrid-ruby.SendGrid.API"))
+              .l
+          ) {
+            case apiClassInit :: apiInit :: Nil =>
+              apiInit.fullName shouldBe s"sendgrid-ruby.SendGrid.API.${Defines.Initialize}"
+              apiClassInit.fullName shouldBe s"sendgrid-ruby.SendGrid.API<class>.${Defines.Initialize}"
```

Comment:
@DavidBakerEffendi can you just double check this? Not sure if it should be `.API<class>.initialize` or `.API.initialize<class>`

Reply:
`.API<class>.initialize` is more correct